modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -80,213 +139,233 @@
-		to_chat(user, span_warning("You start to throw the fuel into the forge..."))
-		if(!do_after(user, 3 SECONDS, target = src)) //need 3 seconds to fuel the forge
-			to_chat(user, span_warning("You abandon fueling the forge."))
-			in_use = FALSE
+		to_chat(user, span_warning("You start to throw [I] into [src]..."))
+		if(!do_after(user, skill_modifier, target = src)) //need 3 seconds to fuel the forge
+			fail_message(user, "You abandon fueling [src].")
 			return
 		var/obj/item/stack/sheet/stack_sheet = I
 		if(!stack_sheet.use(1)) //you need to be able to use the item, so no glue.
-			to_chat(user, span_warning("You abandon fueling the forge."))
-			in_use = FALSE
+			fail_message(user, "You abandon fueling [src].")
 			return
-		forge_fuel_weak += 300 //5 minutes
+		forge_fuel_weak += 5 MINUTES
 		in_use = FALSE
-		to_chat(user, span_notice("You successfully fuel the forge."))
-		if(prob(30))
-			to_chat(user, span_notice("The forge's fuel lights interestingly..."))
-			addtimer(CALLBACK(src, .proc/spawn_coal), 2 MINUTES)
+		to_chat(user, span_notice("You successfully fuel [src]."))
+		user.mind.adjust_experience(/datum/skill/smithing, 3) //useful fueling means you get some experience
+		if(prob(45))
+			to_chat(user, span_notice("[src]'s fuel lights interestingly..."))
+			addtimer(CALLBACK(src, .proc/spawn_coal), 1 MINUTES)
 		return
 
+	//why use coal over wood? the target temp is set to 100 under coal, while only 50 with wood
+	//which means you have to use the billows with wood, but not with coal
 	if(istype(I, /obj/item/stack/sheet/mineral/coal)) //used for strong fuel
+		var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/smithing, SKILL_SPEED_MODIFIER) * 3 SECONDS
 		if(in_use) //only insert one at a time
 			to_chat(user, span_warning("You cannot do multiple things at the same time!"))
 			return
 		in_use = TRUE
-		if(forge_fuel_strong >= 300) //cannot insert too much
-			to_chat(user, span_warning("You only need one to two pieces of coal at a time! You have [forge_fuel_strong] seconds remaining!"))
-			in_use = FALSE
+		if(forge_fuel_strong >= 5 MINUTES) //cannot insert too much
+			fail_message(user, "You only need one piece of coal at a time! You have [forge_fuel_strong/10] seconds remaining!")
 			return
-		to_chat(user, span_warning("You start to throw the fuel into the forge..."))
-		if(!do_after(user, 3 SECONDS, target = src)) //need 3 seconds to fuel the forge
-			to_chat(user, span_warning("You abandon fueling the forge."))
-			in_use = FALSE
+		to_chat(user, span_warning("You start to throw [I] into [src]..."))
+		if(!do_after(user, skill_modifier, target = src)) //need 3 seconds to fuel the forge
+			fail_message(user, "You abandon fueling [src].")
 			return
 		var/obj/item/stack/sheet/stack_sheet = I
 		if(!stack_sheet.use(1)) //need to be able to use the item, so no glue
-			to_chat(user, span_warning("You abandon fueling the forge."))
-			in_use = FALSE
+			fail_message(user, "You abandon fueling [src].")
```

Comment:
for the `!.use` lines (all of them in the files) it might be better for the fail_message to say you FAIL to fuel [src] rather than you ABANDON it. This lets the user know it was a seperate issue, and not just moving.